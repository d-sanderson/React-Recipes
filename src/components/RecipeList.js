import React from 'react';

const RecipeList = (props) => (
    <div style={props.style}>
        <h2>Recipes</h2>
        <ul>
            <li>
                <span>Cupcakes</span>
                <span>Desert</span>
            </li>
            <li>
                <span>Eclair Pie</span>
                <span>Desert</span></li>
            <li>
                <span>Ramen Pot Roast</span>
                <span>Dinner</span>
            </li>
            <li>
                <span>Chicken Pot Pie</span>
                <span>Dinner</span>
            </li>
            <li>
                <span>Chorizo Breakfast Tacos</span>
                <span>Dinner</span>
            </li>
        </ul>
    </div>
)
export default RecipeList