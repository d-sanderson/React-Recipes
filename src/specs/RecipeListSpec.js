import React from 'react';
import renderer from 'react-test-renderer';
import RecipeList from '../components/RecipeList';

const testRecipes = [
  {
    id: 1,
    name: 'Test Recipe',
    category: 'Test Category',
  },
  {
    id: 2,
    name: 'Test Recipe 2',
    category: 'Test Category 2',
  },
];

describe('<RecipeList />', () => {
  test('Should not break when no recipe is passed', () => {
    const component = renderer.create(<RecipeList />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('Should render correctly', () => {
    const component = renderer.create(<RecipeList recipse={testRecipes} />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('Should render favorite state correctly', () => {
    const component = renderer.create(
      <RecipeList recipes={testRecipes} favorited={true} />,
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
