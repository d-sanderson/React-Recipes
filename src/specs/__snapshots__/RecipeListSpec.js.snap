// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> Should correctly render recipe 1`] = `
<li
  className="py2 border-bottom border-bottom-dashed pointer nowrap"
  onClick={[Function]}
>
  <span
    aria-label="favorite"
    className="mr1"
    onClick={[Function]}
    role="img"
  >
    ☠️
  </span>
  <span>
    Test Recipe
  </span>
  <span>
    Test Category
  </span>
</li>
`;

exports[`<Header /> Should not break if no recipe passed 1`] = `
<li
  className="py2 border-bottom border-bottom-dashed pointer nowrap"
  onClick={[Function]}
>
  <span
    aria-label="favorite"
    className="mr1"
    onClick={[Function]}
    role="img"
  >
    ☠️
  </span>
  <span />
  <span />
</li>
`;

exports[`<Header /> Should render favorite state 1`] = `
<li
  className="py2 border-bottom border-bottom-dashed pointer nowrap"
  onClick={[Function]}
>
  <span
    aria-label="favorite"
    className="mr1"
    onClick={[Function]}
    role="img"
  >
    😻
  </span>
  <span>
    Test Recipe
  </span>
  <span>
    Test Category
  </span>
</li>
`;

exports[`<RecipeList /> Should not break when no recipe is passed 1`] = `
<ul
  className="list-reset"
/>
`;

exports[`<RecipeList /> Should render correctly 1`] = `
<ul
  className="list-reset"
/>
`;

exports[`<RecipeList /> Should render favorite state correctly 1`] = `
<ul
  className="list-reset"
>
  <li
    className="py2 border-bottom border-bottom-dashed pointer nowrap"
    onClick={[Function]}
  >
    <span
      aria-label="favorite"
      className="mr1"
      onClick={[Function]}
      role="img"
    >
      😻
    </span>
    <span>
      Test Recipe
    </span>
    <span>
      Test Category
    </span>
  </li>
  <li
    className="py2 border-bottom border-bottom-dashed pointer nowrap"
    onClick={[Function]}
  >
    <span
      aria-label="favorite"
      className="mr1"
      onClick={[Function]}
      role="img"
    >
      😻
    </span>
    <span>
      Test Recipe 2
    </span>
    <span>
      Test Category 2
    </span>
  </li>
</ul>
`;
